oeps/oep-0010.rst

```diff
@@ -0,0 +1,202 @@
+  or might not have a user interface.
+
+- Feature: large applications have major features which are configurable, and
+  should be called out explicitly if their support status is different than the
+  application they are a part of.
+
+
+
+Levels of support
+-----------------
+
+Units in Open edX are either either *included* or *supported* in a release.
+
+To be **included** in an Open edX release, a component, facility or feature
+must meet these criteria.
+
+- It must be successfully merged into its repository's release-from branch.
+  This is typically master, though it can be any branch.
+
+- It must be installable in at least one of the supported installation methods.
+
+- It must be usable in the installed configuration.
+
+Beyond inclusion, to be **supported** in an Open edX release, a component must
+meet further criteria:
+
+- It must be documented thoroughly and accurately.
+
+- An engineering team either in edX or outside of edX, must agree to be
+  minimally available for answering questions in the public channels.
+
+- An engineering team either in edX or outside of edX, must agree to provide
+  fixes if severe problems are identified.
+
+Many of these criteria are open to interpretation, or varying degrees of
+effort. For example, what does "documented thoroughly" mean? We can't quantify
+that. It and other loose criteria are included here because they are important
+parts of providing a finished quality product, and we don't want to overlook
+them.
+
+
+Release creation
+----------------
+
+A new release line is created every four to six months.  Release lines are
+named with words in alphabetical order: Dogwood, Eucalyptus, Ficus, Gingko,
+Hawthorn.  On a release line, there will be a handful of releases:
+Eucalyptus.1, Eucalyptus.2, and so on.
+
+After the .1 release, new releases in a line are made only for severe problems
+such as security problems, data loss, or feature breakage. 
+
+A new release line is created by making a "release master" branch in each
+involved repo.  These are named "open-release/RELEASENAME.master".  This branch
+will be where changes are accumulated to create each release in the line.
+Releases will be tagged "open-release/RELEASENAME.1",
+"open-release/RELEASENAME.2", and so on.
+
+
+Involving repos in the Open edX build process
+---------------------------------------------
+
+:doc:`OEP-2 <oep-0002>` defines a file format for repository metadata.  The 
+``openedx-release`` key is an optional dictionary governing the participation
+of the repo in the Open edX release process.
+
+``openedx-release``: dictionary (optional)
+
+    Possible keys:
+
+    ``ref``
+        The name of the release-from branch in this repo. This is the branch
+        that will be tagged when an Open edX release is made.
+
+    ``requirements``
+        The path to a requirements.txt-style file in this repo. This file will
+        be consulted to find the commit to tag in dependent repos indicated by
```

Comment:
I don't really understand what this means. And looking at https://github.com/edx/edx-platform/blob/master/requirements/edx/github.txt didn't make it any more clear to me.